portfolio/src/main/webapp/script.js

```diff
@@ -98,3 +92,68 @@ function createCmtEl(comment) {
   comElem.appendChild(txtElem);
   return comElem;
 }
+
+/* Post functionality */
+function getPostKey(){
+    let searchParams = (new URL(document.location)).searchParams;
+    let userNum = searchParams.get("post-key");
+    if(postkey == null || postkey.length === 0){
+        return "1";
+    }
+    return postkey;
+}
+
+function getPosts(){
+   //confirm((new URL(document.location)).searchParams);
+   fetch('/get-posts').then(response => response.json()).then((posts) => {
+    const messageBoard = document.getElementById('posts-container');
+    posts.forEach((post) => {
+      messageBoard.appendChild(createCustomPostEl(post));
+    })   });
+    //location.href = '/index.html';
+}
+
+function fetchBlob() {
+  fetch('/blobstore-upload-url').then((response) => {return response.text();}).then((imageUploadUrl) => {
+      const messageForm = document.getElementById('createPost');//this sets the action
```

Comment:
I think something is going wrong with this line. On the deployed app I see this error in the console:
Uncaught (in promise) TypeError: Cannot set property 'action' of null at script.js:123

This means that document.getElementById('createPost') isn't working properly.

Reply:
I think the error is coming from when the onloading of the page happens. The function is called every time the page loads but only has a non-null value when a user is uploading a post. The output given right now is expected given the implementation.